### PR TITLE
Add instrumentation for "Turn into model" option on saved questions

### LINKED ISSUE #31381: Snowplow: Add instrumentation for "Turn into model" option on saved questions
## Context
- [Analytics spec](https://www.notion.so/metabase/Turn-into-model-clicked-a69324cc8e164bd2af85024720a80e43?pvs=4)
- [Slack context](https://metaboat.slack.com/archives/C01LQQ2UW03/p1686064554952519)

### PR DESCRIPTION
**Pull request by oisincoveney**
Closes https://github.com/metabase/metabase/issues/31381

### Description

Adds snowplow schema and event tracking when clicking the `Turn into a model` option when viewing a saved question.

### How to verify
1. Follow [these steps](https://www.notion.so/metabase/Snowplow-integration-5da1f874beda4153b4fccfa6c1e77caa) to setup Snowplow micro locally
2. Open a saved question
3. Go to the options (`...`) and click `Turn into a model`
4. Visit http://localhost:9090/micro/good to see if the event is there

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Review by ranquild — CHANGES_REQUESTED** [review]
Please take a look at the existing schema. The reason for that is how snowplow/redshift integration works, we don't create new schemas per-event, but rather per event groups.

File: snowplow/iglu-client-embedded/schemas/com.metabase/questiontomodel/jsonschema/1-0-0 [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+    "description": "Converting saved questions to models",
+    "self": {
+        "vendor": "com.metabase",
+        "name": "questiontomodel",
+        "format": "jsonschema",
+        "version": "1-0-0"
+    },
+    "type": "object",
+    "properties": {
```
[the remark this answers is not in the document]
Comment (oisincoveney) [reply snowplow/iglu-client-embedded/schemas/com.metabase/questiontomodel/jsonschema/1-0-0:11]: gotcha, I had the impression that the schemas were per-event. Does this change warrant a new version (i.e. a new file with version `1-1-1`?

[the remark this answers is not in the document]
Comment (oisincoveney) [reply snowplow/iglu-client-embedded/schemas/com.metabase/questiontomodel/jsonschema/1-0-0:11]: Ah, and if that's the case, would this mean that we would also change the `trackNewQuestionSaved` schema to `1-1-1` as well?

File: frontend/src/metabase/query_builder/analytics.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ export const trackNewQuestionSaved = (
   createdQuestion,
   isBasedOnExistingQuestion,
 ) => {
-  trackSchemaEvent("question", "1-0-1", {
+  trackSchemaEvent("question", "1-1-1", {
```
Comment (oisincoveney) [inline frontend/src/metabase/query_builder/analytics.js:8]: Followed the same pattern that the `settings` snowplow events have where the schema seems to match the latest version - let me know if this isn't right

File: snowplow/iglu-client-embedded/schemas/com.metabase/question/jsonschema/1-1-1
Comment (WiNloSt) [inline snowplow/iglu-client-embedded/schemas/com.metabase/question/jsonschema/1-1-1:1]: I think you could name the version `1-0-2` instead. This file which I labeled version `1-0-1` also only has an extra event, but not properties.

File: snowplow/iglu-client-embedded/schemas/com.metabase/metabot/jsonschema/1-0-1 [hunk 3]
```diff
@@ -5,7 +5,7 @@
     "vendor": "com.metabase",
     "name": "metabot",
     "format": "jsonschema",
-    "version": "1-0-0"
+    "version": "1-0-1"
```
Comment (WiNloSt) [inline snowplow/iglu-client-embedded/schemas/com.metabase/metabot/jsonschema/1-0-1:8]: This is a drive-by fix I think?

File: frontend/src/metabase/query_builder/analytics.js (hunk 2 above)
Comment (WiNloSt) [reply frontend/src/metabase/query_builder/analytics.js:8]: I think it's okay to update the version where possible, but if we stick to [the semantic versioning](https://www.notion.so/metabase/Snowplow-integration-5da1f874beda4153b4fccfa6c1e77caa?pvs=4#fa3b889daa6949698cc61030f0fcbf2b), `1-1-1` would be incompatible with the previous event since we bump the minor version. It's safe to bump it to `1-0-2` in this case.

**Review by WiNloSt — CHANGES_REQUESTED** [review]
I think overall, this looks great. But I think we should fix the version to follow semantic versioning then we'll be good.

File: snowplow/iglu-client-embedded/schemas/com.metabase/question/jsonschema/1-1-1
Comment (oisincoveney) [reply snowplow/iglu-client-embedded/schemas/com.metabase/question/jsonschema/1-1-1:1]: Gotcha, looking at the criteria for versioning this makes the most sense.

File: snowplow/iglu-client-embedded/schemas/com.metabase/metabot/jsonschema/1-0-1 (hunk 3 above)
Comment (oisincoveney) [reply snowplow/iglu-client-embedded/schemas/com.metabase/metabot/jsonschema/1-0-1:8]: yep

File: frontend/src/metabase/query_builder/components/QuestionActions.tsx [hunk 4]
```diff
@@ -107,6 +108,7 @@ const QuestionActions = ({
     const modal = checkCanBeModel(question)
       ? MODAL_TYPES.TURN_INTO_DATASET
       : MODAL_TYPES.CAN_NOT_CREATE_MODEL;
+    trackTurnIntoModelClicked(question);
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/QuestionActions.tsx:111]: I asked product people about [whether or not this is the button/menu we want to track](https://metaboat.slack.com/archives/C01LQQ2UW03/p1686300137312269?thread_ts=1686064554.952519&cid=C01LQQ2UW03). So let's wait for the confirmation.

Comment (WiNloSt) [reply frontend/src/metabase/query_builder/components/QuestionActions.tsx:111]: Anyhow, even if this is the place we want to track still there should be the following question if we want to track this for both cases
1. When we can actually turn this question into a model
2. When we can't turn this question into a model (as shown in the screenshot below)
image

The second case sounds really odd at least to me. Like we're tracking the menu click even though we know this question couldn't be turned into a model.

Comment (WiNloSt) [reply frontend/src/metabase/query_builder/components/QuestionActions.tsx:111]: Ok so we're tracking the menu click. [Context](https://metaboat.slack.com/archives/C01LQQ2UW03/p1686302270894519?thread_ts=1686064554.952519&cid=C01LQQ2UW03).

**Review by WiNloSt — APPROVED** [review]
Great work!

**Review by ranquild — APPROVED** [review]
